deck/lib/deck/impl.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Deck.Impl do
+
+  # === Public API =========================================================== #
+
+  def new_deck() do
+    values = ~w(2 3 4 5 6 7 8 9 10 J Q K A)
+    suits  = [:spades, :clubs, :hearts, :diamonds]
+    create_cards(values, suits)
+  end
+  
+  def shuffle(deck \\ new_deck()) do
+    Enum.shuffle(deck)
+  end
+  
+  def cut(deck \\ new_deck(), num_cards) do
+    {top, bottom} = Enum.split(deck, num_cards)
+    Enum.concat([bottom, top])
+  end
+  
+  def draw(deck, num_cards) do
+    Enum.split(deck, num_cards)
+  end
+  
+  def burn_card(deck) do
+    {_burned, reduced_deck} = draw(deck, 1);
+    reduced_deck
+  end
+
+  # === Private API ========================================================== #
+  
```

Comment:
You could also do something like

~~~ elixir
iex> values = ~w(2 3 4 5 6 7 8 9 10 J Q K A) |> Stream.cycle
iex> suits  = [:spades, :clubs, :hearts, :diamonds] |> Stream.cycle 
iex> suits |> Enum.take(52) |> Enum.zip(values)
~~~

:)